test/src/main/java/org/apache/accumulo/test/functional/ManagerAssignmentIT.java

```diff
@@ -88,9 +119,283 @@ public void test() throws Exception {
       assertNull(online.future);
       assertNotNull(online.current);
       assertEquals(online.current, online.last);
+
+      // take the tablet offline
+      c.tableOperations().offline(tableName, true);
+      offline = getTabletLocationState(c, tableId);
+      assertNull(offline.future);
+      assertNull(offline.current);
+      assertEquals(flushed.current, offline.last);
+
+      // set the table to ondemand
+      c.tableOperations().onDemand(tableName, true);
+      TabletLocationState ondemand = getTabletLocationState(c, tableId);
+      assertNull(ondemand.future);
+      assertNull(ondemand.current);
+      assertEquals(flushed.current, ondemand.last);
+
+      // put it back online
+      c.tableOperations().online(tableName, true);
+      online = getTabletLocationState(c, tableId);
+      assertNull(online.future);
+      assertNotNull(online.current);
+      assertEquals(online.current, online.last);
+
+    }
+  }
+
+  @Test
+  public void testScannerAssignsOneOnDemandTablets() throws Exception {
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = super.getUniqueNames(1)[0];
+      c.tableOperations().create(tableName);
+      String tableId = c.tableOperations().tableIdMap().get(tableName);
+      loadDataForScan(c, tableName);
+
+      TreeSet<Text> splits = new TreeSet<>();
+      splits.add(new Text("f"));
+      splits.add(new Text("m"));
+      splits.add(new Text("t"));
+      c.tableOperations().addSplits(tableName, splits);
+      c.tableOperations().onDemand(tableName, true);
+      assertTrue(c.tableOperations().isOnDemand(tableName));
+
+      // Wait 2x the TabletGroupWatcher interval for ondemand
+      // tablets to be unassigned.
+      Thread.sleep(10000);
```

Comment:
Instead of sleeping a fixed time could this do

```
while(countHostedTablets() > 0) {
  sleep(50);
}
```

Reply:
Addressed in 8738077